RSSReader/Presenter/FeedPresenter.m

```diff
@@ -31,15 +32,22 @@ - (id)initWithArray:(NSMutableArray *)array networkManager:(NetworkManager *)man
     return self;
 }
 
-- (void)loadNewsWithCompletion:(void (^)(NSError *))completion {
-    [self.networkManager loadFeedWithCompletion:^(NSData * data, NSError *error) {
-        if (error) {
-            completion(error);
-        }
-        else {
-            [self parseRowData:data completion:completion];
-        }
-    }];
+- (void)asyncLoadNewsWithCompletion:(void (^)(NSError *))completion {
+    self.completion = completion;
```

Comment:
It would be better to check completion block against nil prior calling it.

Reply:
fixed